pipelines/k8s-test/Jenkinsfile

```diff
@@ -0,0 +1,250 @@
+                sh "sleep 5m"
+                sh "kubectl logs -l job-name=integration-tests --follow -n test"
+                sh "touch it.log"
+                sh "kubectl logs -l job-name=integration-tests --tail=10000 -n test >> it.log"
+                try {
+                    def failures = sh (script:"cat it.log | grep 'FAILURE!'", returnStdout: true)
+                    if (!failures.isEmpty()){
+                        currentBuild.result = 'FAILED'
+                        currentStage.result = 'FAILURE'
+                    } else {
+                        currentBuild.result = 'SUCCESS'
+                    }
+                } catch (Exception e) {
+                    currentBuild.result = 'FAILED'
+                    currentStage.result = 'FAILURE'
+                }
+                def log_file = sh (script:"kubectl logs -l job-name=integration-tests --tail=2000 -n default | grep maven.out | cut -c 112-300", returnStdout: true)
+                }
+            }
+        }
+    } 
+    catch (any) {
+        // if there was an exception thrown, the build failed
+        currentBuild.result = "FAILED"
+        throw any
+
+    } finally {
+        stage('Delete Cluster') {
+            build job: '../Kubernetes/KubernetesDeploy/azure/master', parameters: [
+                [$class: 'StringParameterValue', name: 'action', value: "destroy"],
+                [$class: 'StringParameterValue', name: 'environment', value: target_cluster],
+                [$class: 'StringParameterValue', name: 'destroy_confirmation', value: target_cluster],
+                [$class: 'StringParameterValue', name: 'kubernetes_version', value: "1.14.8"],
+                [$class: 'StringParameterValue', name: 'enable_cloud_manager', value: "true"],
+                [$class: 'StringParameterValue', name: 'enable_tick', value: "true"],
+                [$class: 'StringParameterValue', name: 'enable_elastic_stack', value: "true"],
+                [$class: 'StringParameterValue', name: 'core_vmsize', value: "Standard_D4s_v3"],
+                [$class: 'StringParameterValue', name: 'core_max_vmcount', value: "3"],
+                [$class: 'StringParameterValue', name: 'db_vmsize', value: "Standard_D16s_v3"],
+                [$class: 'StringParameterValue', name: 'meta_max_vmcount', value: "3"],
+                [$class: 'StringParameterValue', name: 'meta_vmsize', value: "Standard_D8s_v3"],
+                [$class: 'StringParameterValue', name: 'db_max_vmcount', value: "12"],
+                [$class: 'StringParameterValue', name: 'tag_creator', value: scmVars.GIT_BRANCH]
+            ]
+            }
+        }
+        notifyBuild(currentBuild.result)
+}
+
+//scheck for external IP to show and if ingress is accepting connections
+def checkSpliceDB() {
+    sh '''wget http://sourceforge.net/projects/netcat/files/netcat/0.7.1/netcat-0.7.1.tar.gz
```

Comment:
shouldn't this be defined in the requirements e.g. Dockerfile?
`RUN apt-get update && apt-get install -y netcat`

Reply:
I ran into errors where netcat wasn't found in the docker image. I tried putting the package into the docker image as an `apk install`(base is alpine), but I kept getting an error about application not found. I also tried exporting the path inside the docker image without success. The only way I managed to get it working was just to install it as a part of the pipeline as a stage.

Reply:
ok